### PR TITLE
Upgrade terraform-provider-azurerm to v3.45.0

### PR DESCRIPTION
**Pull request by rquitales**

### CODE REVIEW

File: provider/resource.go [hunk 1]
```diff
@@ -301,37 +307,38 @@ func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []str
 // preConfigureCallback returns an error when cloud provider setup is misconfigured
 func preConfigureCallback(vars resource.PropertyMap, c tfshim.ResourceConfig) error {
 
+	// Default environment to Public unless specified.
```
Comment (rquitales) [inline provider/resource.go:310]: New behaviour that should be reviewed.

File: provider/resource.go [hunk 2]
```diff
@@ -301,37 +307,38 @@ func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []str
...
-		ClientID:             stringValue(vars, "clientId", []string{"ARM_CLIENT_ID"}),
-		ClientSecret:         stringValue(vars, "clientSecret", []string{"ARM_CLIENT_SECRET"}),
-		TenantID:             stringValue(vars, "tenantId", []string{"ARM_TENANT_ID"}),
-		Environment:          envName,
-		ClientCertPath:       stringValue(vars, "clientCertificatePath", []string{"ARM_CLIENT_CERTIFICATE_PATH"}),
-		ClientCertPassword:   stringValue(vars, "clientCertificatePassword", []string{"ARM_CLIENT_CERTIFICATE_PASSWORD"}),
-		MsiEndpoint:          stringValue(vars, "msiEndpoint", []string{"ARM_MSI_ENDPOINT"}),
-		AuxiliaryTenantIDs:   auxTenants,
-		ClientSecretDocsLink: "https://www.pulumi.com/docs/intro/cloud-providers/azure/setup/#service-principal-authentication",
+	credentials := auth.Credentials{
```
Comment (rquitales) [inline provider/resource.go:324]: Switching to "github.com/hashicorp/go-azure-sdk/sdk/auth" as the authentication builder we previously used is now empty at the latest version.

File: provider/resource.go (hunk 1 above)
Comment (thomas11) [reply provider/resource.go:310]: Compare with https://github.com/pulumi/pulumi-azuread/pull/351/commits/6d5de8ae14ec345feccc09f1252942fc35dbf5a3#diff-34c57e622183cb0d8dd0d3f9eaa0861b3340120e9b2ad811bac7ac7be4cea4b1 - should be the same logic for consistency.

File: provider/resource.go [hunk 3]
```diff
@@ -301,37 +307,38 @@ func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []str
...
-		SupportsAzureCliToken:          true,
-		SupportsAuxiliaryTenants:       len(auxTenants) > 0,
+		EnableAuthenticatingUsingClientCertificate: true,
+		EnableAuthenticatingUsingClientSecret:      true,
+		EnableAuthenticatingUsingManagedIdentity:   boolValue(vars, "msiEndpoint", []string{"ARM_USE_MSI"}),
+		EnableAuthenticatingUsingAzureCLI:          true,
 	}
 
-	_, err := builder.Build()
+	_, err := auth.NewAuthorizerFromCredentials(context.TODO(), credentials, environment.MicrosoftGraph)
```
Comment (thomas11) [inline provider/resource.go:341]: [nit] `context.Background()` should be fine here, authentication is a long-running background thing

File: provider/resource.go (hunk 1 above)
Comment (rquitales) [reply provider/resource.go:310]: Done! Updated to be the same logic.

**Review by thomas11 — APPROVED** [review]

**Pull request merged**